generators/utilities/src/main/resources/csrc/emulator.cc

```diff
@@ -267,25 +279,25 @@ int main(int argc, char** argv)
 
 #if VM_TRACE
   Verilated::traceEverOn(true); // Verilator must compute traced signals
+#if CY_FST_TRACE
+  std::unique_ptr<VerilatedFstC> tfp(new VerilatedFstC);
+  if (vcdfile_name) {
+    tile->trace(tfp.get(), 99);  // Trace 99 levels of hierarchy
+    tfp->open(vcdfile_name);
```

Comment:
The statements inside the if's here seem very similar can't we just put the different constructors behind the ifdefs and have less repeated code?

Reply:
no. unfortunately, line 286 throws a wrench in that.

Reply:
Why can't we use `tfp->open(vcdfile_name)` for both? Or hoist the vcdfile_name is the FST portion out of the if, like in VCD mode.

Reply:
because, somebody wrote `generators/rocket-chip/src/main/resources/csrc/verilator.h`, which means we have to pass a `FILE` pointer when making a VCD dumper, but the person who wrote the FST extension for verilator didn't use the same paradigm and create a `VerilatedFstFile`, so we can't make a corresponding `VerilatedFstFILE` that takes in a FILE, and all we can work with is a `const char *filename`.

i would have to touch too many things to reduce the code duplication.

Reply:
That somebody is @grebe, who might have a comment. We don't have to use rocket's `verilator.h` technically, but I think in this case it makes sense to.

Reply:
Does this not work?
```
#if CY_FST_TRACE
  std::unique_ptr<VerilatedFstC> tfp(new VerilatedFstC);
#else
  std::unique_ptr<VerilatedVcdFILE> vcdfd(new VerilatedVcdFILE(vcdfile));
  std::unique_ptr<VerilatedVcdC> tfp(new VerilatedVcdC(vcdfd.get()));
#endif // CY_FST_TRACE
  if (vcdfile_name) {
    tile->trace(tfp.get(), 99);  // Trace 99 levels of hierarchy
    tfp->open(vcdfile_name);
  }
```

Reply:
@jwright6323 yep i confirmed that works for both VCD and FST outputs.